FreeRTOS-Plus/Demo/FreeRTOS_Plus_POSIX_with_actor_Windows_Simulator/posix_demo.c

```diff
@@ -370,5 +371,143 @@ void vStartPOSIXDemo( void *pvParameters )
 
 	/* This task was created with the native xTaskCreate() API function, so
 	must not run off the end of its implementing thread. */
+    printf("\n------ SUMMARY ------\n");
+    static char status[40 * 100] = { 0 };
+    vTaskList(status);
+    printf(status);
+    printf("\n");
 	vTaskDelete( NULL );
 }
+
+
+/*-----------------------------------------------------------*/
+#define N_DUMMY_THREADS 10
+#define N_DUMMY_LOOPS   100
+
+static void * prvDummyThread(void* pvArgs)
+{
+    int id = (int)pvArgs;
+
+    struct timespec ts = {
+        .tv_sec = 1,
+        .tv_nsec = 0
+    };
+
+    for (int i = 0; i < N_DUMMY_LOOPS; i++)
+    {
+        printf("Thread[%02d] loop[%02d]\n", id, i);
+        nanosleep(&ts, NULL);
+    }
+
+    while (1);
+
+    return 0;
+}
+
+static void * prvSelfDestructThread(void* pvArgs)
+{
+    struct timespec ts = {
+        .tv_sec = 1,
+        .tv_nsec = 0
+    };
+
+    nanosleep(&ts, NULL);
+    configASSERT(0 == pthread_cancel(pthread_self()));
+
+    ts.tv_sec = 1;
+    while (1) {
+        nanosleep(&ts, NULL);
+    }
+}
+
+/*
+ * Create various joinable/detached threads and cancel/join them in various orders
+ */
+void vTestPthreadCancel(void* pvParameters)
```

Comment:
It's great to include test for any new API. Though please do not mix demo with test. 

We do have tests for POSIX, not here though.

Reply:
Got it. Will remove

Reply:
Is there a more appropriate place to put it, perhaps in some test suite?

Reply:
Let's discuss offline.